### PR TITLE
Fix for zero block index issue of the first stake for the user

### PR DESCRIPTION
**Pull request by pbukva**

### CODE REVIEW

**Comment by ejfitzgerald** [comment]
Remove the build artifacts?

File: contracts/Staking.sol [hunk 1]
```diff
@@ -771,20 +771,21 @@ contract Staking is AccessControl {
...
+        // TODO(pb): Careful: The `StakeCompoundInterest` event doers not carry explicit block number value - it relies
+        //           on the fact that Event implicitly carries value block.number where the event has been triggered,
+        //           what however can be different than value of the `at_block` input parameter passed in.
+        //           Thus this method needs to be EITHER refactored to drop the `at_block` parameter (and so get the
+        //           value internally by calling the `_getBlockNumber()` method), OR the `StakeCompoundInterest` event
+        //           needs to be extended to include the `uint256 sinceBlock` attribute.
+        //           The original reason for passing the `at_block` parameter was to spare gas for calling the
+        //           `_getBlockNumber()` method twice (by the caller of this method + by this method), what might NOT be
+        //           relevant anymore (after refactoring), since caller might not need to use the block number value anymore.
+        emit StakeCompoundInterest(sender, stake.sinceInterestRateIndex, stake.asset.principal, stake.asset.compoundInterest);
```
Comment (pbukva) [inline contracts/Staking.sol:788]: This is the FIX, thus the most important change in this PR

Comment (pbukva) [inline contracts/Staking.sol:788]: This is the FIX, thus the most important change in this PR

Comment (pbukva) [inline contracts/Staking.sol:788]: This is the FIX, thus the most important change in this PR

**Pull request merged**